src/main/java/org/variantsync/diffdetective/analysis/strategies/AnalyzeAndExportIncrementally.java

```diff
@@ -44,42 +48,40 @@ public AnalyzeAndExportIncrementally() {
     }
 
     @Override
-    public void start(Repository repo, Path outputPath, DiffTreeLineGraphExportOptions options) {
-        super.start(repo, outputPath, options);
+    public void start(Repository repo, Path outputPath) {
+        super.start(repo, outputPath);
 
         IO.tryDeleteFile(outputPath);
-        nextChunkToExport = new StringBuilder();
+
+        lineGraphDestination = new ByteArrayOutputStream();
         collectedCommits = 0;
     }
 
     @Override
-    public void onCommit(CommitDiff commit, String lineGraph) {
+    public OutputStream onCommit(CommitDiff commit) {
         ++collectedCommits;
-        nextChunkToExport.append(lineGraph);
-
-        if (collectedCommits >= commitsToExportAtOnce) {
-            exportAppend(outputPath, nextChunkToExport.toString());
-            nextChunkToExport = new StringBuilder();
+        if (collectedCommits > commitsToExportAtOnce) {
+            flush();
             collectedCommits = 0;
         }
+
+        return new CloseShieldOutputStream(lineGraphDestination);
```

Comment:
What's this stream? Can you document why the usage of such a specific stream is ncessary here?

Reply:
The contract between `AnalysisStrategy.onCommit` and its caller states (in the comment of `AnalysisStrategy.onCommit`) that the returned `OutputStream` is closed after the commit has been processed. However, this strategy wants to reuse the same `OutputStream` for multiple Commits, therefore it has to prevent that the caller closes the stream. That's exactly what `CloseShieldOutputStream` does: It overrides the close method, which does nothing, and forwards all other calls to the `lineGraphDestination`.

I add a comment soon.